src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -0,0 +1,904 @@
+        /// <returns>An object that is equivalent to the date contained in s.</returns>
+        public static DateOnly Parse(ReadOnlySpan<char> s) => Parse(s, DateTimeFormatInfo.CurrentInfo, DateTimeStyles.None);
+
+
+        private const ParseFlags ParseFlagsDateMask = ParseFlags.HaveHour | ParseFlags.HaveMinute | ParseFlags.HaveSecond | ParseFlags.HaveTime | ParseFlags.TimeZoneUsed |
+                                                      ParseFlags.TimeZoneUtc | ParseFlags.CaptureOffset | ParseFlags.UtcSortPattern;
+
+        /// <summary>
+        /// Converts a memory span that contains string representation of a date to its DateOnly equivalent by using culture-specific format information and a formatting style.
+        /// </summary>
+        /// <param name="s">The memory span that contains the string to parse.</param>
+        /// <param name="provider">An object that supplies culture-specific format information about s.</param>
+        /// <param name="style">A bitwise combination of enumeration values that indicates the permitted format of s. A typical value to specify is None.</param>
+        /// <returns>An object that is equivalent to the date contained in s, as specified by provider and styles.</returns>
+        public static DateOnly Parse(ReadOnlySpan<char> s, IFormatProvider? provider, DateTimeStyles style = DateTimeStyles.None)
+        {
+            if ((style & ~DateTimeStyles.AllowWhiteSpaces) != 0)
+            {
+                throw new ArgumentException(SR.Argument_InvalidDateStyles, nameof(style));
+            }
+
+            DateTimeResult result = default;
+            result.Init(s);
+
+            if (!DateTimeParse.TryParse(s, DateTimeFormatInfo.GetInstance(provider), style, ref result))
+            {
+                throw new FormatException(SR.Format(SR.Format_BadDateOnly, s.ToString()));
+            }
+
+            if ((result.flags & ParseFlagsDateMask) != 0)
+            {
+                throw new FormatException(SR.Format(SR.Format_DateOnlyContainsNoneDateParts, s.ToString()));
+            }
+
+            return new DateOnly(DayNumberFromDateTime(result.parsedDate));
+        }
+
+        /// <summary>
+        /// Converts the specified span representation of a date to its DateOnly equivalent using the specified format.
+        /// The format of the string representation must match the specified format exactly or an exception is thrown.
+        /// </summary>
+        /// <param name="s">A span containing the characters that represent a date to convert.</param>
+        /// <param name="format">A span containing the characters that represent a format specifier that defines the required format of s.</param>
+        /// <returns>An object that is equivalent to the date contained in s, as specified by format.</returns>
+        public static DateOnly ParseExact(ReadOnlySpan<char> s, ReadOnlySpan<char> format) => ParseExact(s, format, DateTimeFormatInfo.CurrentInfo, DateTimeStyles.None);
```

Comment:
This overload doesn't exist on DateTime.  Why is it needed here?

Reply:
Also, another example where passing in CurrentInfo isn't needed and could instead be null.  This case is also adding unnecessary expense in the case where format is "o" or "r", as the called method is going to overwrite the provider anyway in such cases with Invariant.

Reply:
The goal was not really to have exact parity with DateTime but we expose the APIs that make sense to have. I am seeing having `ParseExact(ReadOnlySpan<char> s, ReadOnlySpan<char> format)` is useful to not force passing the provider all the time when using it. I added this API intentionally. please let me know if you have any concern keeping it. Thanks.

Reply:
Also, I'll clean up all CurrentInfo usage as I indicated in previous comment.

Reply:
> I am seeing having ParseExact(ReadOnlySpan<char> s, ReadOnlySpan<char> format) is useful to not force passing the provider all the time when using it.

If we believe that's important for DateOnly, presumably it would also be important for DateTime, no?  I'm suggesting consistency between the two in this regard is important.  We could, for example, make the corresponding argument to DateTime.ParseExact optional, with a default value of null.

Reply:
You have raised a very good point regarding making the provider parameter as optional. I think that can work here too. For DateOnly/TimeOnly I'll get rid of `ParseExact(ReadOnlySpan s, ReadOnlySpan format)` and I'll make the longer overload take optional paremters. doing that will be similar to DateTime. We can look later if we want to make the DateTime provider parameter as optional too if needed.